### PR TITLE
Added support to send a Telegram notification & a few minor improvements

### PR DESCRIPTION
**Pull request by ethans**
1. Added support to send notification to Telegram
2. Set the paths variable for the "Added Files Hook"
3. Fixed a bug in the hook that failed due to overriding the categories_verbose with the slug
4. Added in the notification hook a couple of environment variable checks before using them (so that it doesn't fail on forks)

During my testing I added a file to the notices folder and then deleted. That's why there are so many writes and deletes. Sorry.

### For Site Improvements

- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by the Bug Alert team if needed.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have run Pelican locally to ensure my changes have not disrupted the look, feel, or functionality of the site.
This worked for me yesterday, but post reboot - I'm unable to reach the site - I get that the connection is refused. But, I didn't make changes in the pelican site.
- [x] I have checked my code and corrected any misspellings.
- [x] I have added a pull request description which details the nature of this change.

Leave the following intact, which notifies our volunteer team: @BugAlertDotOrg/volunteer-team

### CODE REVIEW

File: notice_communication/__main__.py [hunk 1]
```diff
@@ -22,29 +22,44 @@
...
+        if os.getenv('TEXT_EM_ALL_ID'):
+            send_telephony(summary, category, title, url, filename, sms_file_id, phone_file_id)
 
     print("Operations complete.")
 
+def send_telegram(summary, category, title, url):
+    TELEGRAM_API_KEY = os.getenv('TELEGRAM_API_KEY')
+    TG_CHAT_ID = "@BugAlert"
+    msg = f"{title}: {summary}\n{url}"
+    url_to_send = f"https://api.telegram.org/bot{TELEGRAM_API_KEY}/sendMessage?chat_id={TG_CHAT_ID}&text={msg}"
```
Comment (sullivanmatt) [inline notice_communication/__main__.py:59]: I'm floored at how simple this is

File: notice_communication/__main__.py [hunk 2]
```diff
@@ -131,7 +146,7 @@ def get_content_meta(filename):
 
     pattern = "Slug: (.*)"
     groups = re.search(pattern, notice)
-    category_verbose = groups.group(1)
+    slug = groups.group(1)
```
Comment (sullivanmatt) [inline notice_communication/__main__.py:150]: Nice catch

File: .github/workflows/added_posts.yml [hunk 3]
```diff
@@ -6,6 +6,8 @@ on:
   push:
     branches:
       - main
+    paths:
+     - 'content/notices/*'
```
Comment (sullivanmatt) [inline .github/workflows/added_posts.yml:10]: TIL

File: .github/workflows/added_posts.yml [hunk 4]
```diff
@@ -18,6 +20,7 @@ env:
   TWITTER_CONSUMER_KEY: ${{ secrets.TWITTER_CONSUMER_KEY }}
   TWITTER_CONSUMER_SECRET: ${{ secrets.TWITTER_CONSUMER_SECRET }}
   SENDGRID_API_KEY: ${{ secrets.SENDGRID_API_KEY }}
+  TELEGRAM_API_KEY: ${{ secrets.TELEGRAM_API_KEY }}
```
Comment (sullivanmatt) [inline .github/workflows/added_posts.yml:23]: Secret received from @ethans and placed in GitHub.

File: notice_communication/__main__.py [hunk 5]
```diff
@@ -22,29 +22,45 @@
...
+    # Send Telegram
+    if os.getenv('TELEGRAM_API_KEY'):
+        send_telegram(summary, category, title, url)
+
     if os.getenv('TWITTER_BEARER_TOKEN'):
         twitter = get_twitter_client()
         tweet_summary = summary[:220] if len(summary) > 220 else summary
-        tweet = f"{f'{tweet_summary}...'} {url} #BugAlertNotice"
+        ellipsis = "..." if len(summary) > 220 else ""
+        tweet = f"{f'{tweet_summary}{ellipsis}'} {url} #BugAlertNotice"
```
Comment (sullivanmatt) [inline notice_communication/__main__.py:42]: While reviewing this PR I noticed an issue with Twitter formatting that I decided to quickly tackle here.

**Review by sullivanmatt — APPROVED** [review]

**Comment by sullivanmatt** [comment]
I'd feel better getting @mattlorimor's eyes on this too, before merging.

**Review by mattlorimor — APPROVED** [review]

**Pull request merged**